soroban-sdk/src/testutils.rs

```diff
@@ -29,6 +31,83 @@ pub trait Ledger {
         F: FnMut(&mut LedgerInfo);
 }
 
+pub mod budget {
+    use core::fmt::Display;
+
+    #[doc(inline)]
+    pub use crate::env::internal::budget::CostType;
+
+    /// Budget that tracks the resources consumed for the environment.
+    ///
+    /// Note that some budgets – CPU instructions, memory, and VM cost types –
+    /// are likely to be underestimated when running Rust code compared to
+    /// running the WASM equivalent.
+    ///
+    /// ### Examples
+    ///
+    /// ```
+    /// use soroban_sdk::{Env, Symbol};
+    ///
+    /// # #[cfg(feature = "testutils")]
+    /// # fn main() {
+    /// #     let env = Env::default();
+    /// env.budget().reset();
+    /// // ...
+    /// println!("{}", env.budget());
+    /// # }
+    /// # #[cfg(not(feature = "testutils"))]
+    /// # fn main() { }
+    /// ```
+    pub struct Budget(crate::env::internal::budget::Budget);
+
+    impl Display for Budget {
+        fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+            writeln!(f, "CPU Instructions: {}", self.get_cpu_instruction_count())?;
+            writeln!(f, "Memory Bytes: {}", self.get_memory_bytes_count())?;
+            for cost_type in CostType::variants() {
+                writeln!(f, "{cost_type:?}: {}", self.get(*cost_type))?;
+            }
+            Ok(())
+        }
+    }
+
+    impl Budget {
+        pub(crate) fn new(b: crate::env::internal::budget::Budget) -> Self {
+            Self(b)
+        }
+
+        /// Reset the budget.
+        pub fn reset(&mut self) {
+            self.0.reset_default();
+            self.0.reset_unlimited();
+        }
+
+        /// Get the CPU instruction count.
+        ///
+        /// Note that CPU instructions are likely to be underestimated when
+        /// running Rust code compared to running the WASM equivalent.
+        pub fn get_cpu_instruction_count(&self) -> u64 {
+            self.0.get_cpu_insns_count()
+        }
+
+        /// Get the memory bytes used.
+        ///
+        /// Note that memory is likely to be underestimated when running Rust
+        /// code compared to running the WASM equivalent.
+        pub fn get_memory_bytes_count(&self) -> u64 {
+            self.0.get_mem_bytes_count()
+        }
+
+        /// Get other cost counts.
+        ///
+        /// Note that VM cost types are likely to be underestimated when running
+        /// Rust code compared to running the WASM equivalent.
+        pub fn get(&self, cost_type: CostType) -> u64 {
```

Comment:
I would name this `get_input` to be consistent and more explicit, since the input is different from the cost. The comment needs to be updated too.
The input is what goes into the budget model to produce the cost (which has two dimensions  "cpu instructions" and "memory bytes"). For example in a vector clone, the input is the number of elements of the vector and the cost is the cpu instructions and mem bytes allocated to perform the vector clone action.

Reply:
I renamed it to input. Thanks for the explanation, this helped a lot.